### ISSUE TITLE
breaking dependency mismatch

### ISSUE THREAD

**Issue by marcusbesjes** [sequelize/sequelize-auto#242]
following the instructions to simply install `sequelize-auto`, `pg` and `pg-hstore` globally (automatically disregarding any version dependency between them) the `sequalize` module gets some unexpected code from the `pg` module.

running
```
source ./.env && sequelize-auto -h $PG_HOST -d $PG_DBNAME -u $PG_USERNAME -p $PG_PORT  --dialect postgres -o ./src/models/sequalize
```

The error comes from `sequelize/lib/dialects/postgres/connection-manager.js:179:31`:

```
TypeError: connection.query(...).on is not a function
    at /home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/sequelize/lib/dialects/postgres/connection-manager.js:179:31
    at Promise._execute (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/debuggability.js:303:9)
    at Promise._resolveFromExecutor (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/promise.js:483:18)
    at new Promise (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/promise.js:79:10)
    at /home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/sequelize/lib/dialects/postgres/connection-manager.js:178:12
    at PassThroughHandlerContext.finallyHandler (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/finally.js:57:23)
    at PassThroughHandlerContext.tryCatcher (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/home/marcus/.nvm/versions/node/v8.4.0/lib/node_modules/sequelize-auto/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
```

If I look at the line where it's chocking on the type error `TypeError: connection.query(...).on is not a function`, and start logging, I see that `query` is a function looking like this:
```
function (config, values, callback) {
  // can take in strings, config object or query object
  var query
  var result
  if (typeof config.submit === 'function') {
    result = query = config
    if (typeof values === 'function') {
      query.callback = query.callback || values
    }
  } else {
    query = new Query(config, values, callback)
    if (!query.callback) {
      let resolveOut, rejectOut
      result = new Promise((resolve, reject) => {
        resolveOut = resolve
        rejectOut = reject
      })
      query.callback = (err, res) => err ? rejectOut(err) : resolveOut(res)
    }
  }

  if (this.binary && !query.binary) {
    query.binary = true
  }
  if (query._result) {
    query._result._getTypeParser = this._types.getTypeParser.bind(this._types)
  }

  this.queryQueue.push(query)
  this._pulseQueryQueue()
  return result
}
```
which, when I search for `// can take in strings, config object or query object`, turns out to be coming from `pg/lib/client.js:364`.

So to me it looks like there is a dependency here from `sequalize` on `pg` that is not managed, and I'm wondering how I'm supposed to figure out what version to install.

Also I actually don't want to install these tools globally, but locally in a package, so that's where I started, but I ran into the same issue (naturally). I don't understand how, as a user I am supposed to depend on this module.

**Comment by marcusbesjes** [comment]
fix: looking at the `sequelize-auto/node_modules/sequelize/package.json`, I can find 
```
"pg": "^4.4.6"
```
in the devdependencies, based on that I `npm i -g pg@4.4.6` and it runs now so it seems this is the right verson sequalize dependends on.

I don't really agree that it's a dev dependency when I can't use the tool without it.. how is this not just a regular dependency?

**Comment by marcusbesjes** [comment]
OK so it's probably because pg is the postgres specific thing and sequalize can also be used for different databases, so you don't want to install every database connector with it, which would happen if you put them all in dependencies.

Then a way to manage the dependency would be to have npm scripts to e.g. `npm run add-postgres` which would install the correct version for you.

**Comment by devspacenine** [comment]
+1

Any updates on addressing this issue? The pg dependency version has changed again and its a whole major version behind still.

**Comment by philippefutureboy** [comment]
+1

I confirm this is working up to `pg@^6`, a dependency that is 3 years old.
`pg` is now at version 8 - this is definitely an important bump.

@durango: Is this repository dead? Or is there a newer version of this that is included straight in Sequelize?

**Comment by steveschmitt** [comment]
Fixed in `sequelize-auto@0.5.0`

**Comment by philippefutureboy** [comment]
Thanks @steveschmitt 🎉
